File: app/src/main/java/com/example/viper/presentation/modules/rx2interactor/contract/SampleRx2VmContract.java
package com.example.viper.presentation.modules.rx2interactor.contract;


import android.databinding.Bindable;

import java.util.List;

import com.example.viper.data.SampleObject;
import com.vershininds.mixture.presenter.MvpPresenter;
import com.vershininds.mixture.viewmodel.MvpViewModel;


public interface SampleRx2VmContract {

    enum State {
        INITIAL,
        LOADING, EMPTY, DATA,
        ERROR
    }

    interface ViewModel extends MvpViewModel {
        @Bindable
        List<SampleObject> getData();
        void setData(List<SampleObject> data);

        @Bindable
        String getError();
        void setError(String error);

        @Bindable
        State getState();
        void setState(State state);
    }

    interface Presenter extends MvpPresenter<ViewModel> {
        void itemSelected(SampleObject item);
        void retry();
    }
}

File: app/src/main/java/com/vershininds/mixture/sample/presentation/modules/rx2interactor/contract/SampleRx2VmContract.kt
package com.vershininds.mixture.sample.presentation.modules.rx2interactor.contract

import com.vershininds.mixture.action.ViewAction
import com.vershininds.mixture.sample.data.SampleObject
import com.vershininds.mixture.viewmodel.DataModel

interface SampleRx2VmContract {

    sealed class TypeViewAction : ViewAction() {
        class DataAction(val data: DataModel<List<SampleObject>, String>) : TypeViewAction()
        //TODO: create sample show dialog
//        class ErrorDialogWithCustomMessageAction(val errorMsg: String) : TypeViewAction()
    }

    interface ViewModel {
        fun clickOnItem(sampleObject: SampleObject)
        fun retry()
        //TODO: create sample show dialog
//        fun positiveClickErrorDialog()
    }
}